src/client.rs

```diff
@@ -254,3 +254,158 @@ pub struct ClientOptions {
 fn default_local_pow() -> bool {
     true
 }
+
+#[cfg(test)]
+mod tests {
+    use super::{ClientOptionsBuilder, Network};
+
+    #[test]
+    fn single_node_valid_url() {
+        let builder_res = ClientOptionsBuilder::node("https://tangle.iotaqubic.us:14267");
+        assert!(builder_res.is_ok());
+    }
+
+    #[test]
+    fn single_node_invalid_url() {
+        let builder_res = ClientOptionsBuilder::node("some.invalid url");
+        assert!(builder_res.is_err());
+    }
+
+    #[test]
+    fn multi_node_valid_url() {
+        let builder_res = ClientOptionsBuilder::nodes(&["https://tangle.iotaqubic.us:14267"]);
+        assert!(builder_res.is_ok());
+    }
+
+    #[test]
+    fn multi_node_invalid_url() {
+        let builder_res = ClientOptionsBuilder::nodes(&["some.invalid url"]);
+        assert!(builder_res.is_err());
+    }
+
+    #[test]
+    fn multi_node_empty() {
+        let builder_res = ClientOptionsBuilder::nodes(&[]).unwrap().build();
+        assert!(builder_res.is_ok());
+    }
+
+    #[test]
+    fn network_node_empty() {
+        let builder_res = ClientOptionsBuilder::network(Network::Testnet).build();
+        assert!(builder_res.is_ok());
+    }
+
+    #[test]
+    fn single_node_constructor() {
+        let node = "https://tangle.iotaqubic.us:14267";
+        let node_url: url::Url = url::Url::parse(node).unwrap();
+        let client = ClientOptionsBuilder::node(node).unwrap().build();
+        assert_eq!(client.node(), &Some(node_url));
+        assert!(client.nodes().is_none());
+        assert!(client.network().is_none());
+        assert!(client.quorum_size().is_none());
+        assert_eq!(*client.quorum_threshold(), 0);
+    }
+
+    #[test]
+    fn multi_node_constructor() {
+        let nodes = ["https://tangle.iotaqubic.us:14267"];
+        let quorum_size = 5;
+        let quorum_threshold = 0.5;
+        let client = ClientOptionsBuilder::nodes(&nodes)
+            .unwrap()
+            .quorum_size(quorum_size)
+            .quorum_threshold(quorum_threshold)
+            .build()
```

Comment:
Should the builder return an error if there is only one node, but quorum_size = 5?

Reply:
I think it's good to have error returning for each function.

Reply:
Since iota.rs removed quorum, I'm removing this on another PR too.